responsibleai_vision/responsibleai_vision/utils/feature_extractors.py

```diff
@@ -115,3 +85,37 @@ def extract_features(image_dataset: pd.DataFrame,
             row_feature_values.append(image_dataset.iloc[i, j])
         results.append(row_feature_values)
     return results, feature_names
+
+
+def append_exif_features(image, row_feature_values, feature_names,
+                         blacklisted_tags, feature_metadata):
+    if isinstance(image, str):
+        image_pointer_path = get_image_pointer_from_path(image)
+        with Image.open(image_pointer_path) as im:
+            exifdata = im.getexif()
+            for tag_id in exifdata:
+                # get the tag name, instead of human unreadable tag id
+                tag = str(TAGS.get(tag_id, tag_id))
+                data = exifdata.get(tag_id)
+                # decode bytes
+                if isinstance(data, bytes):
+                    data = data.decode()
+                    if len(data) > MAX_CUSTOM_LEN:
+                        data = data[:MAX_CUSTOM_LEN] + '...'
+                if isinstance(data, str):
+                    if tag in feature_names:
```

Comment:
For all conditions involving `in` check, I wonder if the collections objects on the right should be initially casted to sets instead of being lists to ensure O(1) lookup instead of O(n)?

Reply:
I was thinking of that as well - I guess we would need to do some performance testing with a change like that to validate it